engine/src/main/scala/cromwell/engine/io/gcs/GcsBatchCommandContext.scala

```diff
@@ -84,12 +84,27 @@ final case class GcsBatchCommandContext[T, U](request: GcsBatchIoCommand[T, U],
     promise.trySuccess(promiseResponse)
     ()
   }
+  
+  private def handleSuccess(successResult: Either[T, GcsBatchIoCommand[T, U]]) = {
+    val promiseResponse: BatchResponse = successResult match {
+      // Left means the command is complete, so just create the corresponding IoSuccess with the value
+      case Left(responseValue) => Left(success(responseValue))
```

Comment:
ToL: I'll let @danbills or @mcovarr speak with more authority, but I think I've been warned off of using `Either` as a simple type of Coproduct, I think because `Either` is so often used as "`Left`=bad, `Right`=good" that it's confusing to see it used any other way.

Reply:
This predates the use of coproducts in the Codebase and I'm not going to change all that in this PR at least :) But definitely a TechDebt™️ for later if people care about this.